src/AppInstallerTestExeInstaller/main.cpp

```diff
@@ -1,15 +1,106 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+#include <windows.h>
+#include <winreg.h>
+#include <winerror.h>
 #include <iostream>
 #include <fstream>
 #include <filesystem>
 #include <sstream>
 
-// The installer simply prints all args to an output file
+using namespace std::filesystem;
+
+std::wstring registrySubkey = L"SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall\\";
+std::wstring defaultProductID = L"{A499DD5E-8DC5-4AD2-911A-BCD0263295E9}";
+
+std::wstring uninstallWideString;
+std::wstring overriddenProductCode;
+
+void GenerateUninstaller() {
+    path tempPath = temp_directory_path();
+    path uninstallerPath = tempPath;
+    uninstallerPath /= "UninstallTestExe.bat";
+
+    std::cout << "Uninstaller located at path: " << uninstallerPath << '\n';
+    uninstallWideString = uninstallerPath.wstring();
+
+    path uninstallerOutputTextFilePath = tempPath;
+    uninstallerOutputTextFilePath /= "TestExeUninstalled.txt";
+
+    std::ofstream uninstallerScript(uninstallerPath);
+    uninstallerScript << "@echo off\n";
+    uninstallerScript << "ECHO. >" << uninstallerOutputTextFilePath << "\n";
+    uninstallerScript << "ECHO AppInstallerTestExeInstaller.exe uninstalled successfully.\n";
+    uninstallerScript.close();
+}
+
+void WriteToUninstallRegistry() {
+    HKEY hkey;
+    LONG lReg;
+
+    // String inputs to registry must be of wide char type
+    const wchar_t* displayName = L"AppInstallerTestExeInstaller\0";
+    const wchar_t* publisher = L"Microsoft Corporation\0";
+    const wchar_t* uninstallString = uninstallWideString.c_str();
+    DWORD version = 1;
+    DWORD windowsInstaller = 1;
+
+    std::wstring productCode;
+
+    if (!overriddenProductCode.empty()) 

```

Comment:
>overriddenProductCode [](start = 9, length = 21)

Take in value& rather than using globals.

Reply:
removed globals and changed to take in values returned from methods